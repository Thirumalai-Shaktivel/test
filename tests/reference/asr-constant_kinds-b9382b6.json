{
    "basename": "asr-constant_kinds-b9382b6",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/constant_kinds.f90",
    "infile_hash": "c0be463eb7c2d21d76249ef9357c41398b5f61eb3c9fc1821e01469f",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-constant_kinds-b9382b6.stdout",
    "stdout_hash": "8e9f9f2b9c64999fe9bb4ad4fc331d5549b36f77bb23625467d040ba",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}