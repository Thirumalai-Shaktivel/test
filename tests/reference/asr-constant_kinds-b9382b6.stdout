(TranslationUnit (SymbolTable 1 {constant_kinds: (Program (SymbolTable 2 {i1: (Variable 2 i1 Local (ConstantInteger 1 (Integer 4 [])) (ConstantInteger 1 (Integer 4 [])) Parameter (Integer 4 []) Source Public Required .false.), i2: (Variable 2 i2 Local (ConstantInteger 1 (Integer 4 [])) (ConstantInteger 1 (Integer 4 [])) Parameter (Integer 4 []) Source Public Required .false.), i3: (Variable 2 i3 Local (ImplicitCast (ConstantInteger 1 (Integer 8 [])) IntegerToInteger (Integer 4 []) (ConstantInteger 1 (Integer 4 []))) (ConstantInteger 1 (Integer 4 [])) Parameter (Integer 4 []) Source Public Required .false.), r1: (Variable 2 r1 Local (ConstantReal 1.000000 (Real 4 [])) (ConstantReal 1.000000 (Real 4 [])) Parameter (Real 4 []) Source Public Required .false.), r2: (Variable 2 r2 Local (ImplicitCast (ConstantReal 1.000000 (Real 8 [])) RealToReal (Real 4 []) (ConstantReal 1.000000 (Real 4 []))) (ConstantReal 1.000000 (Real 4 [])) Parameter (Real 4 []) Source Public Required .false.), r3: (Variable 2 r3 Local (ConstantReal 1.000000 (Real 4 [])) (ConstantReal 1.000000 (Real 4 [])) Parameter (Real 4 []) Source Public Required .false.), r4: (Variable 2 r4 Local (ImplicitCast (ConstantReal 1.000000 (Real 8 [])) RealToReal (Real 4 []) (ConstantReal 1.000000 (Real 4 []))) (ConstantReal 1.000000 (Real 4 [])) Parameter (Real 4 []) Source Public Required .false.), x: (Variable 2 x Local () () Default (Integer 4 []) Source Public Required .false.), y: (Variable 2 y Local () () Default (Real 4 []) Source Public Required .false.)}) constant_kinds [] [(= (Var 2 y) (ConstantReal 1.000000 (Real 4 [])) () ()) (= (Var 2 y) (ImplicitCast (ConstantReal 1.000000 (Real 8 [])) RealToReal (Real 4 []) (ConstantReal 1.000000 (Real 4 []))) () ()) (= (Var 2 y) (ConstantReal 1.000000 (Real 4 [])) () ()) (= (Var 2 y) (ImplicitCast (ConstantReal 1.000000 (Real 8 [])) RealToReal (Real 4 []) (ConstantReal 1.000000 (Real 4 []))) () ()) (= (Var 2 x) (ConstantInteger 1 (Integer 4 [])) () ()) (= (Var 2 x) (ConstantInteger 1 (Integer 4 [])) () ()) (= (Var 2 x) (ImplicitCast (ConstantInteger 1 (Integer 8 [])) IntegerToInteger (Integer 4 []) (ConstantInteger 1 (Integer 4 []))) () ())])}) [])
